src/ir/memory-utils.cpp

```diff
@@ -62,11 +66,11 @@ bool flatten(Module& wasm) {
     }
     std::copy(segment->data.begin(), segment->data.end(), data.begin() + start);
   }
-  dataSegments.resize(1);
   dataSegments[0]->offset->cast<Const>()->value = Literal(int32_t(0));
   dataSegments[0]->data.swap(data);
   wasm.removeDataSegments(
     [&](DataSegment* curr) { return curr->name != dataSegments[0]->name; });
+  dataSegments.resize(1);
```

Comment:
It looks like moving this would cause problems if there are no segments to begin with. Then `dataSegments[0]` would be out of bounds.

Reply:
Moving this was okay because there is a check above on line 46 to return early `if (dataSegments.size() == 0)`. Removed the resize line and the asan tests are passing 👍🏾